client/my-sites/domains/domain-management/edit-contact-info-page/index.tsx

```diff
@@ -1,19 +1,176 @@
+import ExternalLink from 'calypso/components/external-link';
+import FormattedHeader from 'calypso/components/formatted-header';
 import Main from 'calypso/components/main';
 import BodySectionCssClass from 'calypso/layout/body-section-css-class';
+import { getSelectedDomain } from 'calypso/lib/domains';
+import { localizeUrl } from 'calypso/lib/i18n-utils';
+import Breadcrumbs from 'calypso/my-sites/domains/domain-management/components/breadcrumbs';
+import DomainMainPlaceholder from 'calypso/my-sites/domains/domain-management/components/domain/main-placeholder';
+import NonOwnerCard from 'calypso/my-sites/domains/domain-management/components/domain/non-owner-card';
+import {
+	domainManagementEdit,
+	domainManagementList,
+	domainManagementContactsPrivacy,
+} from 'calypso/my-sites/domains/paths';
 import getCurrentRoute from 'calypso/state/selectors/get-current-route';
 import isRequestingWhois from 'calypso/state/selectors/is-requesting-whois';
+import EditContactInfoFormCard from '../edit-contact-info/form-card';
+import PendingWhoisUpdateCard from '../edit-contact-info/pending-whois-update-card';
+import EditContactInfoPrivacyEnabledCard from '../edit-contact-info/privacy-enabled-card';
+import { EditContactInfoPageProps } from './types';
+
+import './style.scss';
+
+const EditContactInfoPage = ( {
+	currentRoute,
+	domains,
+	isRequestingWhois,
+	selectedDomainName,
+	selectedSite,
+}: EditContactInfoPageProps ): JSX.Element => {
+	const translate = useTranslate();
+
+	const isDataLoading = () => {
+		return ! getSelectedDomain( { domains, selectedDomainName } ) || isRequestingWhois;
+	};
+
+	const goToContactsPrivacy = () => {
+		page( domainManagementContactsPrivacy( selectedSite?.slug, selectedDomainName, currentRoute ) );
+	};
+
+	const renderBreadcrumbs = () => {
+		const previousPath = domainManagementEdit(
+			selectedSite?.slug,
+			selectedDomainName,
+			currentRoute
+		);
+
+		const items = [
+			{
+				label: translate( 'Domains' ),
+				href: domainManagementList( selectedSite?.slug, currentRoute ),
+			},
+			{
+				label: selectedDomainName,
+				href: previousPath,
+			},
+			{ label: translate( 'Edit contact infomation' ) },
+		];
+
+		const mobileItem = {
+			label: translate( 'Back' ),
+			href: previousPath,
+			showBackArrow: true,
+		};
+
+		return <Breadcrumbs items={ items } mobileItem={ mobileItem } />;
+	};
+
+	const renderContent = () => {
+		const domain = getSelectedDomain( { domains, selectedDomainName } );
+
+		if ( ! domain.currentUserCanManage ) {
+			return <NonOwnerCard domains={ domains } selectedDomainName={ selectedDomainName } />;
+		}
+
+		if ( domain.isPendingWhoisUpdate ) {
+			return <PendingWhoisUpdateCard />;
+		}
+
+		if ( domain.mustRemovePrivacyBeforeContactUpdate && domain.privateDomain ) {
+			return (
+				<EditContactInfoPrivacyEnabledCard
+					selectedDomainName={ selectedDomainName }
+					selectedSiteSlug={ selectedSite?.slug as string }
```

Comment:
Since we know that in this case/page/context the `selectedSite` variable cannot be `null`, then I'd suggest using the non-null assertion operator (`!`) for every instance of it - _even though it **is** discouraged by the linter, it is still better/more ascertain than a cast_:
```suggestion
					selectedSiteSlug={ selectedSite!.slug }
```

Reply:
Cool! I was afraid of the linter 😅. Fixed it.